src/Plugins/RpcServer/ParameterConverter.cs

```diff
@@ -0,0 +1,149 @@
+// Copyright (C) 2015-2024 The Neo Project.
+//
+// ParameterConverter.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using Neo.Json;
+using Neo.Plugins.RpcServer.Model;
+using Neo.Wallets;
+using System;
+using System.Collections.Generic;
+using JToken = Neo.Json.JToken;
+
+namespace Neo.Plugins.RpcServer;
+
+public static class ParameterConverter
+{
+    private static readonly Dictionary<Type, Func<JToken, object>> s_conversionStrategies;
+
+    static ParameterConverter()
+    {
+        s_conversionStrategies = new Dictionary<Type, Func<JToken, object>>
+        {
+            { typeof(string), token => Result.Ok_Or(token.AsString, CreateInvalidParamError<string>(token)) },
+            { typeof(byte), ConvertNumeric<byte> },
+            { typeof(sbyte), ConvertNumeric<sbyte> },
+            { typeof(short), ConvertNumeric<short> },
+            { typeof(ushort), ConvertNumeric<ushort> },
+            { typeof(int), ConvertNumeric<int> },
+            { typeof(uint), ConvertNumeric<uint> },
+            { typeof(long), ConvertNumeric<long> },
+            { typeof(ulong), ConvertNumeric<ulong> },
+            { typeof(double), token => Result.Ok_Or(token.AsNumber, CreateInvalidParamError<double>(token)) },
+            { typeof(bool), token => Result.Ok_Or(token.AsBoolean, CreateInvalidParamError<bool>(token)) },
+            { typeof(UInt256), ConvertUInt256 },
+            { typeof(ContractNameOrHashOrId), ConvertContractNameOrHashOrId },
+            { typeof(BlockHashOrIndex), ConvertBlockHashOrIndex }
+        };
+    }
+
+    internal static object ConvertParameter(JToken token, Type targetType)
+    {
+        if (s_conversionStrategies.TryGetValue(targetType, out var conversionStrategy))
+            return conversionStrategy(token);
+        throw new RpcException(RpcError.InvalidParams.WithData($"Unsupported parameter type: {targetType}"));
+    }
+
+    private static object ConvertNumeric<T>(JToken token) where T : struct
+    {
+        if (TryConvertDoubleToNumericType<T>(token, out var result))
+        {
+            return result;
+        }
+
+        throw new RpcException(CreateInvalidParamError<T>(token));
+    }
+
+    private static bool TryConvertDoubleToNumericType<T>(JToken token, out T result) where T : struct
+    {
+        result = default;
```

Comment:
Do we really use floating point numbers anywhere in JSON RPC protocol?

Reply:
It's just kind of dangerous to me since we need precision everywhere.

Reply:
Dangerous only if you use it, this will be updated later. Nothing will be changed in this pr, all logic will follow the old way, having floating check only make this class complete, does not mean it will be used.

Reply:
with `Neo.Json` everything gets set to double anyways. Check the reason why here #3037

Reply:
> having floating check only make this class complete

But why do we need this code if it'll never be used?

Reply:
> > having floating check only make this class complete
> 
> But why do we need this code if it'll never be used?

Cause this is just the first part of the whole task, will keep working on rest rpc methods. Tool class will be kept updating.